app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,109 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.models.Task;
+import com.codesoom.assignment.models.IdGeneratorImpl;
+import org.junit.Assert;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("TaskService 클래스")
+class TaskServiceTest {
+    private static final String TEST_TITLE = "Test Title";
+    private final Task task = new Task(1L, TEST_TITLE);
+    private TaskService taskService;
+
+    @BeforeEach
+    public void setUp() {
+        TaskRepository taskRepository = new TaskRepository(new IdGeneratorImpl());
+        taskRepository.addTask(task);
+        taskService = new TaskService(taskRepository);
+    }
+
+    @Nested
+    @DisplayName("get 메서드들은")
+    class testGet {
+
+        @Nested
+        @DisplayName("식별자 없이 전체 목록을 요청하면")
+        class getWithoutId {
+            @Test
+            @DisplayName("할 일 목록 전체를 반환한다")
+            void getTaskList() {
+                Assert.assertEquals(taskService.getTaskList().size(), 1);
+            }
+        }
+
+        @Nested
+        @DisplayName("식별자를 인자로 전달 받는다면")
+        class getWithId {
+            @Test
+            @DisplayName("식별자에 해당하는 할 일을 반환한다")
+            void getTaskById() {
+                Assertions.assertEquals(TEST_TITLE, taskService.getTaskById(1L).get().getTitle());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("add 메서드는")
+    class testAdd {
+
+        @Nested
+        @DisplayName("할 일을 인자로 받는다면")
+        class addWithParameter {
+            @Test
+            @DisplayName("할 일을 추가하고 추가된 할 일을 반환한다")
+            void addTask() {
+                taskService.addTask(new Task(2L, "Test title2"));
+                Assertions.assertEquals(taskService.getTaskList().size(), 2);
+            }
+        }
+    }
```

Comment:
제가 남긴 주석을 읽어보고 잘 생각해보세요. 테스트 공간을 실험실로 생각해보시면 많은 것을 깨달으실 수 있을 거에요. 실험실을 언급하는 주석은 설명을 위해 남긴 것이니 지우시는 것이 좋아요.

```suggestion
    @Nested
    @DisplayName("add 메서드는")
    class testAdd {
        // 여기는 실험실이 있는 건물입니다.
        // 무엇을 실험하는 실험실인지 간판이 보이는군요. "add 메서드"를 실험하는 곳이네요.

        @Nested
        @DisplayName("할 일이 주어지면")
        class addWithParameter {
            // 여기는 실험을 준비하는 곳입니다. 실험 준비물인 "할 일"(givenTask)을 만들고 있습니다.
            final Task givenTask = new Task(2L, "Test title2");

            @Test
            @DisplayName("할 일을 추가하고 추가된 할 일을 반환한다")
            void addTask() {
                // 여기는 실험실입니다. 여기에서 준비물을 사용해 실험을 진행하고...
                taskService.addTask(givenTask);
                
                // 실험의 결과를 확인합니다.
                Assertions.assertEquals(taskService.getTaskList().size(), 2);
            }
        }
    }
```

Reply:
감사합니다 조금 더 이해가 되는 것 같습니다. 앞으로 반복해서 읽어보겠습니다!